text/1006-deprecate-action-template-helper.md

```diff
@@ -0,0 +1,205 @@
+
+## Summary
+
+The `(action)` template helper and `{{action}}` modifier was common pre-Octane. Now that we have native classes and the `{{on}}` modifier, we no longer need to use `(action)` or `{{action}}`
+
+## Motivation
+
+Remove legacy code with confusing semantics.
+
+This is a part of _[Deprecating Ember Classic (pre-Octane)](https://github.com/emberjs/rfcs/issues/832)_.
+
+## Transition Path
+
+This was written in the [Octave vs Classic cheatsheet](https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/#component-properties__ddau)
+
+<details><summary>that content here</summary>
+
+### Before (pre-Octane)
+
+```js
+// parent-component.js
+import Component from '@ember/component';
+
+export default Component.extend({
+  count: 0
+});
+
+```
+```hbs
+{{!-- parent-component.hbs --}}
+{{child-component count=count}}
+Count: {{this.count}}
+
+```
+```js
+// child-component.js
+import Component from '@ember/component';
+
+export default Component.extend({
+  actions: {
+    plusOne() {
+      this.set('count', this.get('count') + 1);
+    }
+  }
+});
+```
+```hbs
+{{!-- child-component.hbs --}}
+<button type="button" {{action "plusOne"}}>
+  Click Me
+</button>
+```
+
+### After (post-Octane)
+```js
+// parent-component.js
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class ParentComponent extends Component {
+  @tracked count = 0;
+
+  @action plusOne() {
+    this.count++;
+  }
+}
+
+```
+```hbs
+{{!-- parent-component.hbs --}}
+<ChildComponent @plusOne={{this.plusOne}} />
+Count: {{this.count}}
+
+```
+```hbs
+{{!-- child-component.hbs --}}
+<button type="button" {{on "click" @plusOne}}>
+  Click Me
+</button>
+
+```
+
+</details>
+
+But what we could put in the deprecation app:
+
+### Scenario: `action` is passed a string
+
+Before:
+```hbs
+<button type="button" {{action "plusOne"}}>
+  Click Me
+</button>
+```
+
+After
+
+```hbs
+<button type="button" {{on 'click' this.plusOne}}>
+  Click Me
+</button>
+```
+or, if `plusOne` is passed in as an argument 
+```hbs
+<button type="button" {{on 'click' @plusOne}}>
+  Click Me
+</button>
+```
+
+### Scenario: `action` is passed a function reference
+
+Before:
+```hbs
+<SomeComponent @update={{action this.plusOne}} />
+```
+
+After
+
+```hbs
+<SomeComponent @update={{this.plusOne}} />
+```
+
+### Scenario: `action` is passed parameters
+
+Before:
+```hbs
+<SomeComponent @update={{action this.plus 1}} />
+```
+
+After:
+```hbs
+<SomeComponent @update={{fn this.plus 1}} />
+```
+
+### Scenario: `action` is used with `mut` 
+
+Before:
+```hbs
+<SomeComponent @update={{action (mut @value.property}} />
+```
+After:
+```js
+// parent.js
+export default class SomeComponent extends Component {
+    handleUpdate = (value) => this.args.property = value; 
+}
+```
+```hbs
+{{! parent.hbs }}
+<SomeComponent @update={{this.handleUpdate}} />
+```
```

Comment:
This may need to link to the documentation on how to pass actions to components.

I also think we need some mention of action bubbling:
`{{action "foo"}}` on a route template would look first on the controller, then on the route (or perhaps hit one then the other depending on the return value iirc). 

Another thing for a deprecation guide: when moving something out of an actions hash, you need to look for any uses of `send` with the same name and make sure it instead calls the method.

Reply:
added! thanks!